File: app/src/main/java/org/caojun/salmagundi/sharecase/adapter/OrderAdapter.java
package org.caojun.salmagundi.sharecase.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import org.caojun.salmagundi.R;
import org.caojun.salmagundi.sharecase.ormlite.Order;

import java.util.List;

/**
 * Created by CaoJun on 2017/6/14.
 */

/**
 * 订单
 */
public class OrderAdapter extends BaseAdapter {
    private Context context;
    private List<Order> list;

    public OrderAdapter(Context context, List<Order> list) {
        this.context = context;
        setData(list);
    }

    public void setData(List<Order> list) {
        this.list = list;
    }

    @Override
    public int getCount() {
        if(list == null || list.isEmpty()) {
            return 0;
        }
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        if(list == null || list.isEmpty()) {
            return null;
        }
        return list.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if(convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.item_leftright, null);
            holder = new ViewHolder();
            holder.tvLeft = (TextView) convertView.findViewById(R.id.tvLeft);
            holder.tvRight = (TextView) convertView.findViewById(R.id.tvRight);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }
        Order order = (Order) getItem(position);
        //TODO

        return convertView;
    }

    private class ViewHolder {
        TextView tvLeft, tvRight;
    }
}

File: YujiYizidi/src/main/java/org/caojun/yujiyizidi/adapter/OrderAdapter.kt
package org.caojun.yujiyizidi.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import org.caojun.utils.TimeUtils
import org.caojun.yujiyizidi.R
import org.caojun.yujiyizidi.room.Order
import org.caojun.yujiyizidi.room.OrderGoods

/**
 * Created by CaoJun on 2018-1-25.
 */
class OrderAdapter : BaseAdapter {
    private var context: Context? = null
    private val list = ArrayList<Order>()

    constructor(context: Context, list: List<Order>) : super() {
        this.context = context
        setData(list)
    }

    fun setData(list: List<Order>) {
        this.list.clear()
        this.list.addAll(list)
    }

    override fun getView(position: Int, convertView: View?, viewGrouop: ViewGroup?): View {
        var holder: ViewHolder
        var view = convertView
        if (view == null) {
            view = LayoutInflater.from(context).inflate(R.layout.item_order, null)
            holder = ViewHolder()
            holder.tvTime = view?.findViewById(R.id.tvTime)
            holder.tvAmount = view?.findViewById(R.id.tvAmount)
            view?.tag = holder
        } else {
            holder = view.tag as ViewHolder
        }

        val data = getItem(position)
        holder.tvTime?.text = TimeUtils.getTime("yyyy-MM-dd HH:mm:ss", data.time)
        holder.tvAmount?.text = context?.getString(R.string.money, data.amount.toString())
        if (position % 2 == 0) {
            view?.setBackgroundColor(0x33333333)
        } else {
            view?.setBackgroundColor(0xffffff)
        }

        return view!!
    }

    override fun getItem(position: Int): Order = list[position]

    override fun getItemId(position: Int): Long = list[position].id.toLong()

    override fun getCount(): Int = list.size

    private inner class ViewHolder {
        internal var tvTime: TextView? = null
        internal var tvAmount: TextView? = null
    }
}